html/pages/routing/vrf.inc.php

```diff
@@ -1,94 +1,71 @@
 <?php
 
 use LibreNMS\Authentication\Auth;
+use LibreNMS\Exceptions\InvalidIpException;
 
-if (Auth::user()->hasGlobalRead()) {
-    if (!isset($_GET['optb'])) {
-        $_GET['optb'] = 'all';
-    }
-
-    if (!isset($_GET['optc'])) {
-        $_GET['optc'] = 'basic';
-    }
-
-    print_optionbar_start();
-
-    echo '<span style="font-weight: bold;">VRF</span> &#187; ';
-
-    if ($_GET['opta'] == 'vrf' && $_GET['optb'] == 'all') {
-        echo "<span class='pagemenu-selected'>";
-    }
-
-    echo '<a href="routing/vrf/all/'.$_GET['optc'].'/">All</a>';
-    if ($_GET['opta'] == 'vrf' && $_GET['optb'] == 'all') {
-        echo '</span>';
-    }
-
-    echo ' | ';
-    if ($_GET['opta'] == 'vrf' && $_GET['optc'] == 'basic') {
-        echo "<span class='pagemenu-selected'>";
-    }
+if (!Auth::user()->hasGlobalRead()) {
+    include 'includes/error-no-perm.inc.php';
+} else {
+    $link_array = array(
+        'page'     => 'routing',
+        'protocol' => 'vrf',
+    );
 
-    echo '<a href="routing/vrf/'.$_GET['optb'].'/basic/">Basic</a>';
-    if ($_GET['opta'] == 'vrf' && $_GET['optc'] == 'basic') {
-        echo '</span>';
-    }
+    print_optionbar_start('', '');
```

Comment:
You shouldn't need to pass blank params, they have defaults within the called function